[Zhang Xinyi] iP

xyzhang00 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

xyzhang00 committed:
Add basic greet, echo and exit features for Duke

Commit:
Add add and list features to Duke

Commit:
Add the ability to mark tasks as done

Commit:
Add support for tracking three types of tasks:
	- ToDos: tasks without any date/time attached to it e.g., visit new theme park
	- Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
	- Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

Commit:
Edit EXPECTED.txt

Use the input/output redirection technique to semi-automate the testing of Duke

Commit:
Add error handling features to Duke when given unidentified commands or improperly formatted commands

Commit:
Add support for deleting tasks from the list

Commit:
Enable task saving
Add packages

Commit:
Add date and time features

Commit:
Extract Ui, Storage, Parser and TaskList

Commit:
Add JUnit tests

Commit:
Style edits and jar packaging

Commit:
Tweak the code to comply with a coding standard

Tweak the code to comply with a given coding standard

Commit:
Add JavaDoc comments

Add header comments to the non-private class and methods

Commit:
Add find feature

Give users a way to find a task by searching for a keyword

Commit:
Fix branching

Commit:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

* branch-A-CodingStandard:
  Fix branching

Commit:
Fix unit testing

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Fix branching

Commit:
Merge commit 'a75fceeb74708da6ac524c08ef526d38e60ee56b'

* commit 'a75fceeb74708da6ac524c08ef526d38e60ee56b':
  build.gradle: Update version to 8.29
  Add Gradle support

Commit:
Add Gradle support

Commit:
Add Checkstyle support and fix styling errors

Commit:
Add a GUI to Duke
Use the JavaFX technology to implement the GUI

Winniehyx commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,150 @@
+     * Prints given text in output format
+     *
+     * @param text Given text
+     */
```
You can add what this code returns with @return. You can do this for the other methods too.

Winniehyx commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,53 @@
+
+    /**
+     * Gets status
+     * @return Status
```
I think this could be more elaborate

Winniehyx commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,4 @@
-|____/ \__,_|_|\_\___|
-
+    -------------------------------------------->>>>>
+     (≧∇≦)ﾉ Konichiwa~ Watashi wa Duke desu! 
```
I like how you personalise your code

Winniehyx commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,93 @@
+        this.list.add(task);
+    }
+
+    public int size() {
```
This could be renamed to taskSize() since there is already a method called size() in java.

Winniehyx reviewed (commented):
Overall, I found your code clean and easy to read. You followed the coding standards. You can further improve on the javadocs. Good job!

maikongeh commented:
I found you code easy to understand and easy to read. The quality of code was good and you adhered to coding standards. The names of methods were appropriate as well

maikongeh commented on src/main/java/duke/Ui.java:
```diff
@@ -5,24 +5,41 @@
         this.sc = new Scanner(System.in);
     }
 
+    /**
```
I found you code easy to understand and easy to read. The quality of code was good and you adhered to coding standards. The names of methods were appropriate as well. Good work and keep it up!

maikongeh commented on src/main/java/duke/Ui.java:
```diff
@@ -5,24 +5,41 @@
     public void printWelcomeText() {
         System.out.println(divider + space + "Konichiwa~ Watashi wa Duke desu! \n" + space
                 + "What can I do for you today?\n" + divider);
     }
```
I found you code easy to understand and easy to read. The quality of code was good and you adhered to coding standards. The names of methods were appropriate as well. Keep it up

Commit:
Use assertions

Commit:
Improve code quality

Critically examines the code and refactor to improve the code quality where necessary

xyzhang00 committed:
Merge pull request #2 from xyzhang00/A-Assertions

Assertions

xyzhang00 committed:
Merge branch 'master' into A-CodeQuality

xyzhang00 committed:
Merge pull request #1 from xyzhang00/A-CodeQuality

Code quality

Commit:
Add function to easily edit tasks

Support a way to easily edit details of items
Able to update an existing item without having to delete it first

Commit:
Improve GUI and add a User Guide

Commit:
Fix bugs

Commit:
Create JAR file

Commit:
Fix bugs

w-yuchen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String nextLine = sc.nextLine();
+                String[] spiltString = nextLine.split(" \\|\\| ");
+
+                String taskType = spiltString[0];
+                assert taskType.equals("T") || taskType.equals("D") || taskType.equals("E");
+                assert spiltString[1].equals("1") || spiltString[1].equals("0");
+                boolean isCompleted = spiltString[1].equals("1");
+                String taskName = spiltString[2];
+
+                if (spiltString.length == 4) {
+                    LocalDate time = LocalDate.parse(spiltString[3], OUTPUT_FORMATTER);
+                    if (taskType.equals("D")) {
+                        list.add(new Deadline(taskName, isCompleted, time.toString()));
+                    } else if (taskType.equals("E")) {
+                        list.add(new Event(taskName, isCompleted, time.toString()));
+                    } else {
+                        System.out.println("Error");
+                    }
+                } else {
+                    if (taskType.equals("T")) {
+                        list.add(new Todo(taskName, isCompleted));
+                    } else {
+                        System.out.println("Error");
+                    }
+                }
+
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Error!!");
```
This is a rather long method.

w-yuchen commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,197 @@
+        return resultString;
+
+    }
+    public String printNameEdit(Task task, int index) {
+        String resultString = DIVIDER + SPACE + "Done! Name of task number "
+                + index + " has been changed to " + task.getName() + ".\n" + DIVIDER;
+        return resultString;
+    }
+
+    public String printTimeEdit(Task task, int index) {
+        if (task instanceof Todo) {
+            return printErrorText("Sorry! " + task.getTime());
+        } else if (task instanceof Deadline || task instanceof Event) {
+            String resultString = DIVIDER + SPACE + "Done! Time of task number "
+                    + index + " has been changed to " + task.getTime() + ".\n" + DIVIDER;
+            return resultString;
+        } else {
+            return printUnknownInputError();
+        }
+    }
```
I think it might be good to have JavaDocs for these public methods.

w-yuchen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+                boolean isCompleted = spiltString[1].equals("1");
+                String taskName = spiltString[2];
+
+                if (spiltString.length == 4) {
+                    LocalDate time = LocalDate.parse(spiltString[3], OUTPUT_FORMATTER);
+                    if (taskType.equals("D")) {
+                        list.add(new Deadline(taskName, isCompleted, time.toString()));
+                    } else if (taskType.equals("E")) {
+                        list.add(new Event(taskName, isCompleted, time.toString()));
+                    } else {
+                        System.out.println("Error");
+                    }
+                } else {
+                    if (taskType.equals("T")) {
+                        list.add(new Todo(taskName, isCompleted));
+                    } else {
+                        System.out.println("Error");
+                    }
+                }
```
It could be possible to extract this part out to shorten this method.

w-yuchen reviewed (commented):
As this is a really late review, your code is already in a very polished state, so it looks very good and I am just nitpicking here and there.
